core/trino-main/src/main/java/io/trino/sql/analyzer/Analysis.java

```diff
@@ -1210,6 +1211,22 @@ public List<TableInfo> getReferencedTables()
                 .collect(toImmutableList());
     }
 
+    private static Optional<Expression> resolveColumnMask(QualifiedName tableName, String fieldName, Map<Field, Expression> expressions)
+    {
+        return expressions.entrySet().stream()
+                .filter(fieldExpression -> fieldExpression.getKey().canResolve(concatIdentifier(tableName, fieldName)))
+                .findFirst()
```

Comment:
Is it ok to have multiple matches here? Shouldn't we expect `Optional` semantic here?

Reply:
This is a good question. I think it's impossible to have multiple matches here. Like @martint says, ideally this would be a `Map` lookup, this should be a 1:0..1 relationship here.

Reply:
So let's use `.collect(toOptional())`

Reply:
Done